sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -832,6 +832,14 @@ object SQLConf {
       .intConf
       .createWithDefault(10000)
 
+  val IGNORE_DATA_LOCALITY =
+    buildConf("spark.sql.sources.ignore.datalocality")
+      .doc("If it is set to true, Spark will not fetch the block locations for each file on " +
+        "listing files, which will greatly accelerate the list file operation, while loss " +
+        "data locality.")
```

Comment:
Shall we add `.internal()` since this has a drawback, too?

Reply:
I think the doc could be a little bit more explicit about the cases this is useful, how about:

If true, Spark will not fetch the block locations for each file.  This speeds up file listing, but the scheduler cannot schedule tasks to take advantage of data locality.  It can be particularly useful if data is read from a remote cluster so the scheduler could never take advantage of locality anyway.

I don't agree with this logic for making it internal.  Pretty much every conf has some drawback (otherwise we wouldn't have a conf, we'd just permanently change behavior).  That said, maybe its OK to leave it internal as I still feel like we should have some more general solution, as you might have a mix of local and remote filesystems.